crates/pallet-domains/src/staking.rs

```diff
@@ -0,0 +1,61 @@
+//! Staking for domains
+
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use sp_domains::{DomainId, EpochIndex, ExecutorPublicKey};
+use sp_runtime::Percent;
+use sp_std::collections::btree_map::BTreeMap;
+use sp_std::vec::Vec;
+
+/// Type that represents an operator pool details.
+#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
+pub struct OperatorPool<Balance, NominatorId> {
+    pub signing_key: ExecutorPublicKey,
+    pub current_domain_id: DomainId,
+    pub minimum_nominator_stake: Balance,
+    pub nomination_tax: Percent,
+    /// Total active stake for the current pool.
+    pub current_total_stake: Balance,
+    /// Total stake for the current pool in the next epoch.
+    pub next_total_stake: Balance,
+    /// Total shares of the nominators and the operator in this pool.
+    pub total_shares: Balance,
+    pub is_frozen: bool,
+    /// Nominators under this operator pool.
+    pub nominators: BTreeMap<NominatorId, Nominator<Balance>>,
+    /// Pending transfers that will take effect in the next epoch.
+    pub pending_transfers: Vec<PendingTransfer<NominatorId, Balance>>,
+}
+
+/// Type that represents an nominator details under a specific operator pool
+#[derive(TypeInfo, Debug, Encode, Decode, Clone, PartialEq, Eq)]
+pub struct Nominator<Balance> {
+    pub shares: Balance,
```

Comment:
IIUC `shares` is not the amount of staking fund but instead some kind of factor for calculating `fund = shares/total_shares * total_pool_staking_fund`, in this case using `Balance` as its type seems a bit inappropriate.

Reply:
No, that is incorrect. Do take a look at updated V2 spec for detailed explanation

Reply:
I take a closer look at the updated staking protocol, I think my understanding is correct, and my concern just the same as https://github.com/subspace/subspace/pull/1629#discussion_r1254961370